### PR TITLE
Move to pkg:dart_flutter_team_lints

### PR DESCRIPTION
**Pull request by kevmoo**
Some non-trivial, but nice cleanup in lib/src/sdk.dart

### CODE REVIEW

File: lib/src/sdk.dart [hunk 1]
```diff
@@ -282,6 +273,14 @@ class _DownloadedFlutterSdk {
         .transform<String>(utf8.decoder)
         .listen((string) => stderr.write(string));
 
-    return await process.exitCode;
+    final code = await process.exitCode;
+    if (throwOnError && code != 0) {
```
Comment (kevmoo) [inline lib/src/sdk.dart:277]: this just seems better all around

File: analysis_options.yaml [hunk 2]
```diff
@@ -13,18 +13,11 @@ analyzer:
     - lib/src/protos/**
     - local_pub_cache/**
     - project_templates/**
+  errors:
+    lines_longer_than_80_chars: ignore
 
 linter:
   rules:
-    - always_declare_return_types
```
Comment (kevmoo) [inline analysis_options.yaml:19]: Removed duplicative lints

**Comment by kevmoo** [comment]
CC @parlough

File: tool/fuzz_driver.dart [hunk 3]
```diff
@@ -32,9 +32,9 @@ analysis_server.AnalysisServerWrapper? analysisServer;
 
 late comp.Compiler compiler;
 
-var random = Random(0);
-var maxMutations = 2;
-var iterations = 5;
+Random random = Random(0);
```
Comment (domesticmouse) [inline tool/fuzz_driver.dart:35]: I'm curious why you are adding types here, but losing them elsewhere

**Review by domesticmouse — APPROVED** [review]

File: tool/fuzz_driver.dart (hunk 3 above)
Comment (kevmoo) [reply tool/fuzz_driver.dart:35]: that IS weird. It's just the tooling. I'll look quick.

Comment (parlough) [reply tool/fuzz_driver.dart:35]: I believe this comes from the [`type_annotate_public_apis`](https://dart.dev/tools/linter-rules#type_annotate_public_apis) lint which prefers explicit type annotations on top-level variables.

**Pull request merged**